ambry-router/src/main/java/com/github/ambry/router/PutOperation.java

```diff
@@ -1613,6 +1649,13 @@ int fillFrom(ByteBuf channelReadBuf) {
     void checkAndMaybeComplete() {
       boolean done = false;
       // Now, check if this chunk is done.
+      if (chunkException != null && chunkException.getErrorCode() == RouterErrorCode.BlobCorrupted) {
+        logger.error("{} : Put chunk {} failed due to corruption. Failing entire operation", loggingContext,
+            getChunkBlobId());
+        setOperationExceptionAndComplete(chunkException);
```

Comment:
could you please add a comment here. Usually we would set the chunk blob id to null before setting operation exception, but since we want to delete this chunk id later so we won't set it to null and just have it added to metadata chunk so later it will be removed by put manager.

Reply:
Thank you for pointing this! Actually I think `chunkBlobId` is set to `null` when we call `setOperationExceptionAndComplete(chunkException)`. To clean this particular chunk from servers, I reused the slipped put list now. The rest of the chunks (if this is a composite blob) seem to be already cleared in  `PutManager#onComplete()`. What do you think?